posthog/tasks/update_cache.py

```diff
@@ -91,17 +91,24 @@ def update_cache_item(key: str, cache_type: CacheType, payload: dict) -> None:
     cache.set(key, {"result": result, "type": cache_type, "last_refresh": timezone.now()}, CACHED_RESULTS_TTL)
 
     dashboard_items.update(last_refresh=timezone.now(), refreshing=False)
+    if result:
+        cache.set(key, {"result": result, "type": cache_type, "last_refresh": timezone.now()}, CACHED_RESULTS_TTL)
+        return result
+    return None
 
 
 def update_dashboard_items_cache(dashboard: Dashboard) -> None:
     for item in DashboardItem.objects.filter(dashboard=dashboard, filters__isnull=False).exclude(filters={}):
         update_dashboard_item_cache(item, dashboard)
 
 
-def update_dashboard_item_cache(dashboard_item: DashboardItem, dashboard: Optional[Dashboard]) -> None:
+def update_dashboard_item_cache(
+    dashboard_item: DashboardItem, dashboard: Optional[Dashboard]
+) -> Optional[List[Dict[str, Any]]]:
     cache_key, cache_type, payload = dashboard_item_update_task_params(dashboard_item, dashboard)
-    update_cache_item(cache_key, cache_type, payload)
+    result = update_cache_item(cache_key, cache_type, payload)
     dashboard_item.refresh_from_db()
+    return update_cache_item(cache_key, cache_type, payload)
```

Comment:
i think result goes here, otherwise the update function is being triggered twice